core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3894,6 +3894,14 @@ public static List compact(List list) {
     return result;
   }
 
+  /** Support the ARRAY_APPEND function. */
+  public static List arrayAppend(List list, Object element) {
+    final List result = new ArrayList(list.size() + 1);
```

Comment:
Can you resolve it?

![image](https://github.com/apache/calcite/assets/45968640/bee8250d-cf12-423c-9d24-757816750069)